src/Codeception/Module/Filesystem.php

```diff
@@ -75,6 +79,9 @@ protected function absolutizePath(string $path): string
      */
     public function openFile(string $filename): void
     {
+        if (!file_exists($this->absolutizePath($filename))) {
+            TestCase::fail('file not found');
+        }
         $this->file = file_get_contents($this->absolutizePath($filename));
```

Comment:
avoid double evaluating `$this->absolutizePath($filename)`. Store its value in a variable instead.